webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/uploading/UploadService.scala

```diff
@@ -141,9 +147,47 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
         f"Reserving dataset upload of ${reserveUploadInformation.organization}/${reserveUploadInformation.name} with id ${reserveUploadInformation.uploadId}...")
     } yield ()
 
+  def addUploadIdsToOngoingUploads(ongoingUploadsWithoutIds: List[OngoingUpload]): Fox[List[OngoingUpload]] =
+    for {
+      maybeOngoingUploads <- Fox.sequence(
+        ongoingUploadsWithoutIds.map(
+          upload =>
+            runningUploadMetadataStore
+              .find(Json.stringify(Json.toJson(upload.dataSourceId)))
+              .map(uploadIdOpt => uploadIdOpt.map(uploadId => upload.copy(uploadId = uploadId)))
+        ))
+      foundOngoingUploads = maybeOngoingUploads.filter(idBox => idBox.isDefined).flatMap(idBox => idBox.value).collect {
+        case Some(value) => value
+      }
```

Comment:
This seems convoluted. what is the type of maybeOngoingUploads? I think flatten/flatMap may already do what you need?

Reply:
I found a more consice version: 
```scala
foundOngoingUploads = maybeOngoingUploads.collect { case Full(Some(value)) => value }
```

The type of `maybeOngoingUploads` is `List[Box[Option[OngoingUpload]]]`